README.md

```diff
@@ -50,10 +50,18 @@ Package versions and targets may be specified using the `--version` and `--targe
 To install an unsupported crate, you may specify the Cargo.toml metadata entries for `pkg-url`, `bin-dir`, and `pkg-fmt` at the command line, with values [as documented below](#supporting-binary-installation).
 
 For example:
-```
+```shell
 $ binstall \
   --pkg-url="{ repo }/releases/download/{ version }/{ name }-{ version }-{ target }.{ archive-format }" \
   --pkg-fmt="txz" crate_name
+
+$ # fnm
+$ cargo binstall fnm --pkg-url="https://github.com/Schniz/fnm/releases/download/v1.31.1/fnm-linux.zip" \
```

Comment:
```suggestion
$ cargo binstall fnm --pkg-url="https://github.com/Schniz/fnm/releases/download/v{ version }/fnm-linux.zip" \
```

Use `{ version }` here so that this command would work for any version of `fnm`.

Reply:
> Use `{ version }` here so that this command would work for any version of `fnm`.

Wow!, this looks like magic.
So the binstall fetch the package metadata beforehand  from crates.io?

Reply:
> So the binstall fetch the package metadata beforehand from crates.io?

Yes, but the version is determined by retrieving metadata of the crate from crates.io, then use version_req to find the matched ones.

We download the tarball of the crate from crates.io mainly to determine the names of the binaries, section `package.metadata.binstall` for overrides of the default and the repository, which is used to generate the `pkg-url`.

Reply:
@azzamsa Check [SUPPORT.md](https://github.com/cargo-bins/cargo-binstall/blob/main/SUPPORT.md) for how you can use `package.metadata.binstall` to override default.

That might be useful if you want fnm and broot to have out-of-the-box support for cargo-binstall.

Reply:
> @azzamsa Check [SUPPORT.md](https://github.com/cargo-bins/cargo-binstall/blob/main/SUPPORT.md) for how you can use `package.metadata.binstall` to override default.

Yes. I read the template here over and over again.

Reply:
@azzamsa Note that we also support override per target basis:

```
[package.metadata.binstall.overrides.x86_64-pc-windows-msvc]
pkg-fmt = "zip"
pkg-url = "..."
bin-dir = "..."
```

So for fnm, you can provide only `[package.metadata.binstall.overrides.x86_64-unknown-linux-gnu]` as it only supports that target.

For `broot`, the `pkg-url` is simply `ttps://github.com/Canop/broot/releases/download/v{ version }/broot_{ version }.zip` for all target and `pkg-fmt` is `"zip"`, but the `bin-dir` should be `{ name }-{ version }/{ target }/{ bin }{ binary-ext }`.